lib/pure/ioselects/ioselectors_select.nim

```diff
@@ -314,7 +314,7 @@ proc selectInto*[T](s: Selector[T], timeout: int,
 
   if timeout != -1:
     when defined(genode) or defined(freertos) or defined(zephyr) or defined(nuttx):
-      tv.tv_sec = Time(timeout div 1_000)
+      tv.tv_sec = posix.Time(timeout div 1_000)
```

Comment:
Does this imply a breaking change?

Reply:
Generics completely ignore ambiguous types in the current compiler, the `Time` identifier here is actually ambiguous between `times.Time` and `posix.Time` but only by chance the compiler picked `posix.Time` first and captured it. It now gives an ambiguity error, the same one it would give if the proc wasn't generic.

It's breaking in the sense that it can cause compile errors but only for code that essentially worked by random chance. I also think all it does would be causing compile errors, I don't think there's a chance of it leaking into runtime or anything like that. An exception is macros having to deal with symchoices but it really shouldn't be a problem, maybe we can still mention the AST difference in the changelog just in case someone encounters it.

Reply:
Added to changelog